api/soap/mc_filter_api.php

```diff
@@ -147,3 +191,165 @@ function mc_filter_get_issue_headers( $p_username, $p_password, $p_project_id, $
+ * @return array that represents an IssueHeaderDataArray structure
+ */
+function mc_filter_search_issue_headers( $p_username, $p_password, $p_filter_search ) {
+
+    global $_SOAP_API_TO_FILTER_API_NAMES;
+
+    $t_user_id = mci_check_login( $p_username, $p_password );
+
+    if( $t_user_id === false ) {
+        return mci_soap_fault_login_failed();
+    }
+
+    // object to array
+    if( is_object($p_filter_search) ) {
+        $p_filter_search = get_object_vars($p_filter_search);
+    }
+
+    $t_project_id = $p_filter_search['project_id'];
+
+    if( !mci_has_readonly_access( $t_user_id, $t_project_id  ) ) {
+        return mci_soap_fault_access_denied( $t_user_id );
+    }
+
+    $t_filter = array( '_view_type' => 'advanced' );
+
+    // default fields
+    foreach( $_SOAP_API_TO_FILTER_API_NAMES as $t_soap_name => $t_filter_name ) {
+        if ( isset ( $p_filter_search[$t_soap_name]) ) {
+
+            $t_value = $p_filter_search[$t_soap_name];
+            error_log('Simple extraction, value is ' . print_r($t_value, true) . ' .' );
+            $t_filter[$t_filter_name] = $t_value;
+        }
+    }
+
+    // custom fields
+    if( isset ( $p_filter_search['custom_fields']) ) {
+        foreach(  $p_filter_search['custom_fields'] as $t_custom_field ) {
+
+            // object to array
+            if( is_object($t_custom_field) ) {
+                $t_custom_field = get_object_vars($t_custom_field);
+            }
+
+            // if is set custom_field's id, use it primary
+            if ( isset($t_custom_field['id']) ) {
+                $t_custom_field_id = $t_custom_field['id'];
+            }
+            else {
+                $t_custom_field_id = custom_field_get_id_from_name( $t_custom_field['name'] );
+            }
+
+            $t_value = $t_custom_field['value'];
+            error_log('custom field id ' . $t_custom_field_id . print_r($t_value, true) . ' .' );
+            $t_filter['custom_fields'][$t_custom_field_id] =  $t_value;
+        }
+    }
+
+    $t_filter = filter_ensure_valid_filter( $t_filter );
+
+    $t_result = array();
+    $t_page_number = 0;
+    $t_per_page = $p_filter_search['issues_per_page'];
+    $t_page_count = 0;
+    $t_bug_count = 0;
+    $t_rows = filter_get_bug_rows( $t_page_number, $t_per_page, $t_page_count, $t_bug_count, $t_filter );
+
+    foreach( $t_rows as $t_issue_data ) {
+        $t_result[] = mci_issue_data_as_header_array( $t_issue_data );
+    }
+
+    return $t_result;
+}
+
+/**
+ * Get all issues matching the custom filter.
+ *
+ * @param string                $p_username         The name of the user trying to access the filters.
+ * @param string                $p_password         The password of the user.
+ * @param FilterSearchData      $p_filter_search    The custom filter.
+ * @return array that represents an IssueDataArray structure
+ */
+function mc_filter_search_issues( $p_username, $p_password, $p_filter_search ) {
```

Comment:
Looks like we have too much code duplication between the two methods.  How about the following:
- Adding mc_issue_get_many( array(ids) ) that returns an array of the same element type as mc_issue_get().
- Using the output of the issue_search_headers() to provide the input for mc_issue_get_many().

Suggests for better name for mc_issue_get_many() is certainly welcome!

In some cases, this may mean an extra round trip, but it is useful to have our methods usable in more than one scenario.  We can even go to the model where we just return ids from the search and then have methods than return array of headers or array of full issues.